internal/driver/readhandler.go

```diff
@@ -24,23 +46,86 @@ func (d *Driver) HandleReadCommands(deviceName string, protocols map[string]mode
 
 	d.Logger.Debugf("Driver.HandleReadCommands: protocols: %v resource: %v attributes: %v", protocols, reqs[0].DeviceResourceName, reqs[0].Attributes)
 
-	// create device client and open connection
-	endpoint, err := config.FetchEndpoint(protocols)
+	availableServerEndpoints, err := opcua.GetEndpoints(d.serviceConfig.OPCUAServer.Endpoint)
 	if err != nil {
-		return nil, err
+		d.Logger.Error("OPC GetEndpoints: %w", err)
+	}
+	credentials, err := getCredentials(d.serviceConfig.OPCUAServer.CredentialsPath)
+	if err != nil {
+		d.Logger.Error("getCredentials: %w", err)
+	}
```

Comment:
Same as above. Why not just
```suggestion
	if err != nil {
		d.Logger.Error("getCredentials: %w", err)
		return nil, err
	}
```

Reply:
done